integration-tests/src/tests/client/features/flat_storage.rs

```diff
@@ -0,0 +1,132 @@
+use crate::tests::client::process_blocks::deploy_test_contract_with_protocol_version;
+use crate::tests::client::runtimes::create_nightshade_runtimes;
+use near_chain::ChainGenesis;
+use near_chain_configs::Genesis;
+use near_client::test_utils::TestEnv;
+use near_crypto::{InMemorySigner, KeyType, Signer};
+use near_primitives::test_utils::encode;
+use near_primitives::transaction::{Action, ExecutionMetadata, FunctionCallAction, Transaction};
+use near_primitives::version::ProtocolFeature;
+use near_primitives_core::config::ExtCosts;
+use near_primitives_core::hash::CryptoHash;
+use near_primitives_core::types::Gas;
+use nearcore::config::GenesisExt;
+
+/// Check that after flat storage upgrade:
+/// - value read from contract is the same;
+/// - touching trie node cost for read decreases to zero.
+#[test]
+fn test_flat_storage_upgrade() {
+    let mut genesis = Genesis::test(vec!["test0".parse().unwrap(), "test1".parse().unwrap()], 1);
+    let epoch_length = 12;
+    let old_protocol_version = ProtocolFeature::FlatStorageReads.protocol_version() - 1;
```

Comment:
super nit: I find the following easier to understand and requiring fewer arithmetic operations :)
```
let new_protocol_version = ProtocolFeature::FlatStorageReads.protocol_version();
let old_protocol_version = new_protocol_version - 1;
```